File: 01_Calorie_Counting/01.py
INPUT = "input"


def get_calories():
    with open(INPUT, "r") as file:
        calories = [x.split() for x in file.read().strip().split("\n\n")]
        calories = [[int(y) for y in x] for x in calories]
    return calories


def main():
    calories = get_calories()
    sum_calories = list(map(sum, calories))
    max_calories = max(sum_calories)
    print(max_calories)
    
    sorted_sum_calories = sorted(sum_calories, reverse=True)
    print(sum(sorted_sum_calories[:3]))


if __name__ == "__main__":
    main()

File: aoc_2022_elixir/01_Calorie_Counting/01.exs
{:ok, file} = File.read("./input")

calories =
  file
  |> String.trim()
  |> String.split("\n\n")
  |> Enum.map(&String.split/1)
  |> Enum.map(fn x -> Enum.map(x, &String.to_integer/1) end)

calories_sums = Enum.map(calories, &Enum.sum/1)

Enum.max(calories_sums)
|> IO.inspect()

Enum.sort(calories_sums)
|> Enum.take(-3)
|> Enum.sum()
|> IO.inspect()
